### ISSUE TITLE
[bitnami/minio] Fails to start when release-name != minio

### ISSUE THREAD

**Issue by kfox1111** [bitnami/charts#9013]
### Name and Version

bitnami/minio 10.1.1

### What steps will reproduce the bug?

helm upgrade --install testothernameminio bitnami/minio -f /tmp/foo.yaml  --set auth.rootUser=$(uuidgen) --set auth.rootPassword=$(uuidgen) --version 10.1.1

### What do you see instead?

accesskey/secret doesn't get properly into ~/.mc/config

### Additional information

[kfox@zathras ~]$ kubectl logs testothernameminio-78c8ff44c8-6t5nx -f
 19:42:11.02 
 19:42:11.02 Welcome to the Bitnami minio container
 19:42:11.02 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-minio
 19:42:11.02 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-minio/issues
 19:42:11.02 
 19:42:11.02 INFO  ==> ** Starting MinIO setup **
minio 19:42:11.03 INFO  ==> Starting MinIO in background...
minio 19:42:21.05 INFO  ==> Adding local Minio host to 'mc' configuration...
minio 19:42:21.10 INFO  ==> MinIO is already stopped...

**Comment by kfox1111** [comment]
I had a second minio running with the name minio in the same namespace that was working fine when I added the second one. I deleted the pod for the 'minio' instance and it no longer starts too. So the presence of the otherly named minio causes issues too.

**Comment by kfox1111** [comment]
If I clear it out / make a new namespace, then install just testothernameminio it still fails the same way.

So, feels like one of the environment variables maybe that mess with it. Here's a diff between a working and nonworking env:
```
--- /tmp/a	2022-02-14 11:45:50.898015639 -0800
+++ /tmp/b	2022-02-14 11:46:42.496490242 -0800
@@ -16,7 +16,7 @@
 EUID=1001
 GROUPS=()
 HOME=/
-HOSTNAME=minio-58665589f6-jtgmh
+HOSTNAME=testothernameminio-67cbc77f67-d89qn
 HOSTTYPE=x86_64
 IFS=$' \t\n'
 KUBERNETES_PORT=tcp://10.96.0.1:443
@@ -62,5 +62,18 @@
 SHELLOPTS=braceexpand:hashall:interactive-comments
 SHLVL=1
 TERM=xterm
+TESTOTHERNAMEMINIO_PORT=tcp://10.97.78.28:9000
+TESTOTHERNAMEMINIO_PORT_9000_TCP=tcp://10.97.78.28:9000
+TESTOTHERNAMEMINIO_PORT_9000_TCP_ADDR=10.97.78.28
+TESTOTHERNAMEMINIO_PORT_9000_TCP_PORT=9000
+TESTOTHERNAMEMINIO_PORT_9000_TCP_PROTO=tcp
+TESTOTHERNAMEMINIO_PORT_9001_TCP=tcp://10.97.78.28:9001
+TESTOTHERNAMEMINIO_PORT_9001_TCP_ADDR=10.97.78.28
+TESTOTHERNAMEMINIO_PORT_9001_TCP_PORT=9001
+TESTOTHERNAMEMINIO_PORT_9001_TCP_PROTO=tcp
+TESTOTHERNAMEMINIO_SERVICE_HOST=10.97.78.28
+TESTOTHERNAMEMINIO_SERVICE_PORT=9000
+TESTOTHERNAMEMINIO_SERVICE_PORT_MINIO_API=9000
+TESTOTHERNAMEMINIO_SERVICE_PORT_MINIO_CONSOLE=9001
 UID=1001
 _=bash
```

```

**Comment by kfox1111** [comment]
Ok. If I just deploy a release name 'minio' with nothing else in the namespace, I can break it by defining: 
```
export TESTOTHERNAMEMINIO_PORT=tcp://10.97.78.28:9000
```

This doesn't break it:
```
export TESTOTHERNAMEMINIO_PORT_9000_TCP=tcp://10.97.78.28:9000
```
So some script is choking on variables that end with "MINIO_PORT" as a var name.

**Comment by kfox1111** [comment]
Actually, now I can't reproduce the issue anymore. Just starting completely from scratch:
```
helm upgrade --install minio bitnami/minio --set auth.rootUser=d3ac7a0b-c882-44ab-8a94-bad1a90f9dc2 --set auth.rootPassword=d3ac7a0b-c882-44ab-8a94-bad1a90f9dc2 --version 10.1.1
```
Is now failing the same way:
```
$ kubectl logs deploy/minio -f
 21:30:24.53 
 21:30:24.53 Welcome to the Bitnami minio container
 21:30:24.53 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-minio
 21:30:24.53 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-minio/issues
 21:30:24.53 
 21:30:24.53 INFO  ==> ** Starting MinIO setup **
minio 21:30:24.55 INFO  ==> Starting MinIO in background...
minio 21:30:34.58 INFO  ==> Adding local Minio host to 'mc' configuration...
minio 21:30:34.63 INFO  ==> MinIO is already stopped...

```

**Comment by kfox1111** [comment]
Yeah. fresh minikube and fresh minio are crashing this way right now. so can't debug further until I can figure out whats going wrong. the Initial issue I've seen on multiple clusters. so that is real still.

**Comment by kfox1111** [comment]
Seeing it on chart 10.1.7 as well.

**Comment by rafariossaa** [comment]
Hi,
  I was not able to reproduce the issue. I tried this in minikube:
```
$ helm install myminio -f values.yaml .  --set auth.rootUser=rootuser --set auth.rootPassword=myrootpassword --version 10.1.1
$ helm install testminio -f values.yaml .  --set auth.rootUser=rootuser --set auth.rootPassword=myrootpassword --version 10.1.1
$ helm install testminiolastversion -f values.yaml .  --set auth.rootUser=rootuser --set auth.rootPassword=myrootpassword
$ helm install testminio2 -f values.yaml .  --set auth.rootUser=$(uuidgen) --set auth.rootPassword=$(uuidgen)
 
```
And I got all the pods up and running.

```
$ $ kubectl get pods
NAME                                    READY   STATUS    RESTARTS   AGE
myminio-8669f8cd4c-kdv4f                1/1     Running   0          4m9s
testminio-844bbd7bbc-br4lm              1/1     Running   0          3m30s
testminio2-67cc96dbc6-gc74n             1/1     Running   0          27s
testminiolastversion-774c4fb59f-prvnt   1/1     Running   0          3m7s
```

**Comment by rafariossaa** [comment]
Could you indicate witch kubernetes cluster are you using ? I was not able to reproduce the issue in GCE nor minikube.

**Comment by masterkain** [comment]
I'm on microk8s and seeing this

```
│  07:59:20.63                                                                                                                                                                                                                                                                                                          ││  07:59:20.63 Welcome to the Bitnami minio container                                                                                                                                                                                                                                                                   ││  07:59:20.63 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-minio                                                                                                                                                                                                                 ││  07:59:20.64 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-minio/issues                                                                                                                                                                                                             ││  07:59:20.64                                                                                                                                                                                                                                                                                                          ││  07:59:20.64 INFO  ==> ** Starting MinIO setup **                                                                                                                                                                                                                                                                     ││ minio 07:59:20.65 INFO  ==> Starting MinIO in background...                                                                                                                                                                                                                                                           ││ minio 07:59:30.68 INFO  ==> Adding local Minio host to 'mc' configuration...                                                                                                                                                                                                                                          ││ minio 07:59:30.74 INFO  ==> MinIO is already stopped...                                                                                                                                                                                                                                                               ││ Stream closed EOF for drone-cache/drone-cache-minio-84fb56c7-spk52 (minio)
```

```
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: drone-cache
  namespace: drone-cache
spec:
  releaseName: drone-cache
  chart:
    spec:
      chart: minio
      version: "11.1.3"
      sourceRef:
        kind: HelmRepository
        name: bitnami
        namespace: flux-system
  interval: 1h0m0s
  install:
    remediation:
      retries: 4
  values:
    auth:
      rootUser: drone
      rootPassword: drone
    defaultBuckets: icoretech-k8s-drone
    extraEnv:
      - name: MINNIO_LOG_LEVEL
        value: DEBUG
      - name: MINIO_LOG_LEVEL
        value: DEBUG
```

**Comment by rafariossaa** [comment]
Hi,
  Could you try to install it by using the following ?
```
$ helm install drone-cache bitnami/minio
NAME: drone-cache
LAST DEPLOYED: Thu Mar 17 10:08:43 2022
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
CHART NAME: minio
CHART VERSION: 11.1.2
APP VERSION: 2022.3.14
...
```
  I could see it was deployed without issues:
```
$ kubectl get pods
NAME                                 READY   STATUS    RESTARTS   AGE
drone-cache-minio-5bcb75df85-lfr9j   1/1     Running   0          3m12s
```

```
$ kubectl logs drone-cache-minio-5bcb75df85-lfr9j 
 09:09:12.04 
 09:09:12.04 Welcome to the Bitnami minio container
 09:09:12.04 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-minio
 09:09:12.04 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-minio/issues
 09:09:12.04 
 09:09:12.05 INFO  ==> ** Starting MinIO setup **
minio 09:09:12.07 INFO  ==> Starting MinIO in background...
minio 09:09:17.12 INFO  ==> Adding local Minio host to 'mc' configuration...
minio 09:09:17.20 INFO  ==> Stopping MinIO...
 09:09:18.26 INFO  ==> ** MinIO setup finished! **

minio 09:09:18.29 INFO  ==> ** Starting MinIO **
API: http://172.17.0.4:9000  http://127.0.0.1:9000 

Console: http://172.17.0.4:9001 http://127.0.0.1:9001 

Documentation: https://docs.min.io

 You are running an older version of MinIO released 2 days ago 
 Update: Run `mc admin update` 

```
